pulsar-broker/src/test/java/org/apache/pulsar/compaction/CompactionTest.java

```diff
@@ -1967,6 +1967,15 @@ public void testCompactionDuplicate() throws Exception {
         // Wait for phase one to complete
         Thread.sleep(500);
 
+        Awaitility.await().untilAsserted(() -> {
+            PersistentTopic persistentTopic = (PersistentTopic) pulsar.getBrokerService().getTopicReference(topic).get();
+            PulsarTopicCompactionService topicCompactionService =
+                    (PulsarTopicCompactionService) persistentTopic.getTopicCompactionService();
+            Optional<CompactedTopicContext> compactedTopicContext = topicCompactionService.getCompactedTopic()
+                    .getCompactedTopicContext();
+            Assert.assertTrue(compactedTopicContext.isPresent());
+        });
+
```

Comment:
I think it should be moved to after topic unload.

Reply:
has been fixed!